framework/include/systems/NonlinearEigenSystem.h

```diff
@@ -146,6 +164,29 @@ class NonlinearEigenSystem : public NonlinearSystemBase
    */
   TagID nonEigenMatrixTag() { return _A_tag; }
 
+  /**
+   * If the preconditioning matrix includes eigen kernels
+   */
+  void precondMatrixIncludesEigenKernels(bool precond_matrix_includes_eigen)
+  {
+    _precond_matrix_includes_eigen = precond_matrix_includes_eigen;
+  }
+
+  bool precondMatrixIncludesEigenKernels() { return _precond_matrix_includes_eigen; }
+
+  TagID precondMatrixTag() { return _precond_tag; }
+
+  virtual void attachMoosePreconditioner(Preconditioner<Number> * preconditioner) override;
+
+  Preconditioner<Number> * moosePreconditioner() { return _moose_preconditioner; }
```

Comment:
Breaking encapsulation by returning pointer to non-const, but I guess you need to do this?

Reply:
I will try if I can add `const` .

A good practice is to add `const` to everything and then start to remove that if it is inappropriate

Reply:
We need to call non const member function during PCApply so we can not add const to the return value